server/src/main/java/org/elasticsearch/index/codec/vectors/es818/DirectIOLucene99FlatVectorsFormat.java

```diff
@@ -67,50 +61,13 @@ public FlatVectorsWriter fieldsWriter(SegmentWriteState state) throws IOExceptio
         return new Lucene99FlatVectorsWriter(state, vectorsScorer);
     }
 
-    private static final IOContext DIRECT_IO_CONTEXT = new IOContext() {
-        @Override
-        public Context context() {
-            return Context.DEFAULT;
-        }
-
-        @Override
-        public MergeInfo mergeInfo() {
-            return null;
-        }
-
-        @Override
-        public FlushInfo flushInfo() {
-            return null;
-        }
-
-        @Override
-        public Set<FileOpenHint> hints() {
-            return Set.of(DirectIOHint.INSTANCE);
-        }
-
-        @Override
-        public IOContext withHints(FileOpenHint... hints) {
-            return this;
-        }
-
-        @Override
-        public Optional<ReadAdvice> readAdvice() {
-            return Optional.empty();
-        }
-
-        @Override
-        public IOContext withReadAdvice(ReadAdvice advice) {
-            return this;
-        }
-    };
-
     @Override
     public FlatVectorsReader fieldsReader(SegmentReadState state) throws IOException {
         SegmentReadState directIOState = new SegmentReadState(
             state.directory,
             state.segmentInfo,
             state.fieldInfos,
-            DIRECT_IO_CONTEXT,
+            state.context.withHints(DirectIOHint.INSTANCE),
```

Comment:
I used a specific implementation of `IOContext` so the hints weren't overridden by calls to `withHints` within this method. I should have commented that.

Reply:
Ah, we have just discovered that other directory implementations can be used. So DirectIOHint should probably be added to all the others - we can modify our implementation here to add `DirectIO` to any `withHints` that are specified